board/rockchip/evb_rk3399/evb-rk3399.c

```diff
@@ -30,3 +38,75 @@ int board_early_init_f(void)
 	return 0;
 }
 #endif
+
+
+
+/* ALT_INFO Macro Function below added for DFU_ALT_INFO */
+
+#ifdef CONFIG_SET_DFU_ALT_INFO
+
+#define CONFIG_DFU_ALT_BOOT_SD \
+	"mmc 1=idbloader-img raw 0x40 0x2000;u-boot-itb raw 0x4000 0x2000\0"
+
+#define CONFIG_DFU_ALT_BOOT_EMMC \
+	"mmc 0=idbloader-img raw 0x40 0x2000;u-boot-itb raw 0x4000 0x2000\0"
+
+char *get_dfu_alt_boot(char *interface, char *devstr)
+{
+	char *alt_boot;
+	struct mmc *mmc;
+	int dev_num = 0;
+
+	if (devstr == NULL) {
+		devstr = "0";
+	}
+	alt_boot = env_get("dfu_alt_boot");
+
+	if (alt_boot == NULL) {
+		dev_num = simple_strtoul(devstr, NULL, 10);
+		mmc = find_mmc_device(dev_num);
+		if (!mmc) {
+			return NULL;
+		}
+		if (mmc_init(mmc)) {
+			return NULL;
+			}
+		if (IS_SD(mmc)) {
+			alt_boot = CONFIG_DFU_ALT_BOOT_SD;
+		} else {
+			alt_boot = CONFIG_DFU_ALT_BOOT_EMMC;
+		}
```

Comment:
Nit: the coding standard is a bit off here. I would write the above hunk like this:

```
	if (!alt_boot) {
		if (!mmc || mmc_init(mmc))
			return NULL;
		alt_boot = IS_SD(mmc) ? CONFIG_DFU_ALT_BOOT_SD : CONFIG_DFU_ALT_BOOT_EMMC;
	}
```

Reply:
Modified